abdullahbayram/shopping-basket-rn-js
# Feature/add unit tests for molecules

### Summary
This pull request adds unit tests for molecule components and introduces a reusable testing utility for rendering components wrapped in a ThemeProvider.

### Changes
- Added unit tests for the following molecule components:
  - ActivityOverlay
  - CheckoutCard
  - Toggle
- Improved unit tests for the following molecule components:
  - Header
  - Input
  - ProductCard

- Introduced `renderInThemeProvider` utility function for wrapping components in tests.
- Updated ESLint configuration to disable `react/jsx-props-no-spreading` for `*.test.js` files.

### How to Test
1. Pull the branch locally: `git checkout feature/add-unit-tests-for-molecules`
2. Run tests: `npm run test`
3. Verify all tests pass and snapshots are consistent.

### Checklist
- [x] Unit tests added for new components.
- [x] Code linted and formatted.
